geode-core/src/test/java/org/apache/geode/TX2JUnitTest.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode;
+
+import static org.apache.geode.distributed.ConfigurationProperties.MCAST_PORT;
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.CacheFactory;
+import org.apache.geode.cache.CacheTransactionManager;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.distributed.internal.DistributionConfig;
+import org.apache.geode.internal.cache.GemFireCacheImpl;
+import org.apache.geode.internal.cache.TXManagerImpl;
+import org.apache.geode.internal.cache.TXStateProxyImpl;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+import org.apache.logging.log4j.Logger;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.experimental.categories.Category;
+
+@Category(IntegrationTest.class)
+public class TX2JUnitTest {
+
+  private static final Logger logger = LogService.getLogger();
+
+  @Rule
+  public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+
+  long k1 = 1;
+  long k2 = 2;
+  long k3 = 3;
+  long k4 = 4;
+  String v1 = "value1";
+  String v2 = "value2";
+  String v3 = "value3";
+  String regionName = "region1";
+  Cache cache;
+
+  enum SetOp {
+    KEYSET, VALUES, ENTRYSET;
+  }
+
+  @After
+  public void tearDownTXJUnitTest() throws Exception {
+    closeCache();
+  }
+
+  @Test
+  public void testValuesCallStartsTx() {
+    verifySetOp(SetOp.VALUES, false);
```

Comment:
Does this style of test bother others as much as it does me? To me the test method itself should be telling you what is happening and what is expected. Here, we have two arbitrary values (the enum and boolean) that determine what happens multiple calls deep in a call stack. It is very hard to reason on what each test does: for each test, you need to go back through the private methods and see what their particular values cause to happen.

Personally, I would prefer the tests be written more like:
```
@Test
public void myTest() {
  doGenericTestSetup();
  doAction1();
  doAction2();
  for (value v : values) {
    assertDidTheRightThing();
  }
}
```
It is ok for test code in my opinion to be a little less DRY in order for the tests to be clear in what they are testing.